src/pages/index.js

```diff
@@ -62,36 +104,128 @@ function handleImageClick(cardData) {
 
 //edit profile
 function handleProfileSubmit(data) {
-  userInfo.setUserInfo({name: data.name, job: data.description});
-editPopup.close(); }
-
-//add card
+  editPopup.renderLoading(true);
+  api.updateUserInfo({
+    name: data.name,
+    about: data.about
+  })
+    .then(updatedUserData => {
+      user.setUserInfo({
+        name: updatedUserData.name, 
+        about: updatedUserData.about || updatedUserData.dascription || updatedUserData.bio
+      });
+    })
+    .catch(err => console.error(`Error updating profile: ${err}`))
+    .finally(() =>{
+      editPopup.renderLoading(false);
+      editPopup.close();
+    }
+    )
+   }
+
+// Add card
 function handleAddCardSubmit(data) {
-  const card = {
+  const cardData = {
     name: data.title,
     link: data.url
   };
-  const cardElement = createCard(card);
-  section.addItems(cardElement);
-  addcardPopup.close();
-  profileAddFormElement.reset();
-  addFormValidator.disableButton();
+  addCardPopup.renderLoading(true);
+  // Create the Card 
+  api.addNewCard(cardData)
+    .then((newCardData) => {
+      const cardElement = createCard(newCardData);
+      section.addItems(cardElement);  
+      addCardPopup.close();            
+      profileAddFormElement.reset();  
+      addFormValidator.disableButton();  
+    })
+    .catch(err => {
+      console.error(`Error adding card: ${err}`);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    }
+    )
+}
+
+
+
+//avatar
+function handleAvatarSubmit(data){
+  avatarPopup.renderLoading(true);
+  api.updateUserAvatar({
+    avatar: data.url
+  })
+    .then(updateUserData => {
+      user.setUserInfo({
+        avatar: updateUserData.avatar
+      });
+      
+    })
+    .catch(err => console.error(`Error updating avatar: ${err}`))
+    .finally(() =>{
+      avatarPopup.renderLoading(false);
+      avatarPopup.close();
+      avatarFormElement.reset();
+    });
 }
 
-function openEditProfileModal() {
-  const currentUser = userInfo.getUserInfo();
+//delete
+function handleDeleteSubmit(card){
+  deletePopup.setSubmitFunction(() => {
+    deletePopup.renderLoading(true);
+    api.deleteCard(card._id)
+      .then(() => {
+        card.removeCard();
+      })
+      .catch(err => {
+        console.error(`Error deleting card ${err}`);
+      })
+      .finally(()=>{
+        deletePopup.renderLoading(false);
+        deletePopup.close();
+      })
+  });
+  deletePopup.open();  
+} 
+    
+  
+
+//like
+function handleCardLike(card){
+  const isLiked = card.isLiked(); 
+
+  const likePromise = isLiked
+    ? api.removeLike(card.getId())  
+    : api.addLike(card.getId()); 
+
+  likePromise
+    .then(updatedCardData => {
+      card.updateLikes(updatedCardData.isLiked);  
+    })
+    .catch(err => console.error(`Error updating like status: ${err}`));
+}
+
+function openModal(popup) {
```

Comment:
`openModal` can't be used for the avatar modal because you insert the `profile data` inside it. It's not needed on avatar modal opening